app/src/main/java/org/mozilla/focus/search/SearchEngineListPreference.java

```diff
@@ -40,24 +39,24 @@ public SearchEngineChooserPreference(Context context, AttributeSet attrs, int de
     protected View onCreateView(ViewGroup parent) {
         final View layoutView = super.onCreateView(parent);
         searchEngineGroup = layoutView.findViewById(R.id.search_engine_group);
-        searchEngineGroup.setOnCheckedChangeListener(this);
         refreshSearchEngines();
         return layoutView;
     }
 
-    @Override
-    public void onCheckedChanged (RadioGroup group, int checkedId) {
-        Settings.getInstance(group.getContext()).setDefaultSearchEngine(searchEngines.get(checkedId));
-    }
+    protected abstract int getItemResId();
+    protected abstract void updateDefaultItem(CompoundButton defaultButton);
 
     public void refreshSearchEngines() {
         if (searchEngineGroup == null) {
             // There is no search engine group yet.
             return;
         }
-        final SearchEngineManager sem = SearchEngineManager.getInstance();
         final Context context = searchEngineGroup.getContext();
         final Resources resources = context.getResources();
+        final SearchEngineManager sem = SearchEngineManager.getInstance();
+
+        // Force SearchEngineManager to refetch, to get the newest search engine.
+        sem.init(context);
```

Comment:
Those init calls are problematic. SearchEngineManager registers a broadcast receiver and now this happens multiple times. Previously this happened only for the Application context (as init was called from FocusApplication). Now we register it multiple times for the Activity, which at some point will get finished. But the registration still exists and keeps a reference to the Activity causing:

> android.app.IntentReceiverLeaked: Activity org.mozilla.focus.activity.SettingsActivity has leaked IntentReceiver org.mozilla.focus.search.SearchEngineManager@5f00ea4 that was originally registered here. Are you missing a call to unregisterReceiver()?

There's no need to register/unregister new receivers, so I guess it would be enough to call loadSearchEngines() instead of init(). Also we might want to use the Application Context as the load operation on a background thread might again live longer than the activity.